### PR TITLE
Feature/add ulx timedowoify

### PR DESCRIPTION
**Pull request by Claydo9**
Add a timed ULX command that allows staff member to curse a player and ulx owoify them at random intervals for a random duration.

### CODE REVIEW

File: lua/ulx/modules/sh/timed_owoify.lua [hunk 1]
```diff
@@ -0,0 +1,49 @@
+CFCUlxCommands.timedtrainfuck = CFCUlxCommands.timedtrainfuck or {}
```
Comment (Periapsises) [inline lua/ulx/modules/sh/timed_owoify.lua:1]: Should this be kept here?
If yes then I think it's the wrong name for it

File: lua/ulx/modules/sh/timed_owoify.lua [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    local function newDelay()
+        return math.Rand( 5, 25 * 60 )
+    end
+    
+    local function newDuration()
+        return math.Rand( 5, 5 * 60 )
+    end 
+
+    local function enable( caller, ply )
+        local name = timerName( ply )[1]
```
Comment (Periapsises) [inline lua/ulx/modules/sh/timed_owoify.lua:20]: This function returns a vararg which cannot be indexed like a table
```suggestion
        local name = timerName( ply )
```

File: lua/ulx/modules/sh/timed_owoify.lua [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+        timer.Create( name, newDelay(), 0, function()
+            if not IsValid( ply ) then
+                return timer.Remove( name )
+            end
+            CFCUlxCommands.owoify.owoifyCommand( caller, ply, false )
+            timer.Adjust( timerName( ply )[2] ), newDuration() )
```
Comment (Periapsises) [inline lua/ulx/modules/sh/timed_owoify.lua:27]: Same thing here
```suggestion
            timer.Adjust( select( 2, timerName( ply ) ) ), newDuration() )
```

File: lua/ulx/modules/sh/timed_owoify.lua [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        end )
+        timer.Create( timerName( ply )[2], newDuration(), 0, function()
+            local t_name = timerName( ply )[2]
```
Comment (Periapsises) [inline lua/ulx/modules/sh/timed_owoify.lua:30]:
```suggestion
            local _, t_name = timerName( ply )
```

File: lua/ulx/modules/sh/timed_owoify.lua [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+            if not IsValid( ply ) then
+                return timer.Remove( t_name )
+            end
+            CFCUlxCommands.owoify.owoifyCommand( caller, ply, true )
+            timer.Adjust( timerName( ply )[1], newDelay() )
```
Comment (Periapsises) [inline lua/ulx/modules/sh/timed_owoify.lua:35]:
```suggestion
            local adjustedTimerName = timerName( ply )
            timer.Adjust( adjustedTimerName , newDelay() )
```

File: lua/ulx/modules/sh/timed_owoify.lua [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        end )
+    end
+
+    local function disable( ply )
+        timer.Remove( timerName( ply )[1] )
+        timer.Remove( timerName( ply )[2] )
```
Comment (Periapsises) [inline lua/ulx/modules/sh/timed_owoify.lua:41]: Same as the above comments, ideally you could localize both values with one call
```lua
local timerName1, timerName2 = timerName( ply )
timer.Remove( timerName1 )
timer.Remove( timerName2 )
```

**Comment by Claydo9** [comment]
The timers in this code automatically start and will cause chaos, currently figuring out how to resolve this but am at a bit of a roadblock so I will draft it for now.

**Comment by legokidlogan** [comment]
What if it was a random chance to owoify any particular message, instead of toggling over intervals?

**Pull request closed without merging**